### ISSUE TITLE
[ISSUE] No output with embulk 0.8.1 without `-X min_output_tasks=1`

### ISSUE THREAD

**Issue by sonots** [embulk/embulk#381]
I found this issue with

* https://github.com/sonots/embulk-filter-column
* https://github.com/sonots/embulk-filter-row

(More plugins would exist)

HOW TO REPRODUCE:

```
git clone https://github.com/sonots/embulk-filter-column
cd embulk-filter-column
./gradlew classpath
embulk run -I lib example/columns.yml # NO OUTPUT!!!!
embulk run -I lib example/columns.yml -X min_output_tasks=1 # Get output
```

They worked nicely with 0.7.9.

**Comment by civitaspo** [comment]
I found too
- https://github.com/civitaspo/embulk-filter-to_json

HOW TO REPRODUCE(In my plugin):
```
git clone https://github.com/civitaspo/embulk-filter-to_json.git
cd embulk-filter-to_json
./gradlew classpath
embulk run -I lib example/config.yml # NO OUTPUT!!!!
embulk run -I lib example/config.yml -X min_output_tasks=1 # Get output
```

I think this happen when `min_output_tasks` is larger than input tasks.

**Comment by frsyuki** [comment]
Do you know why this happens?

**Comment by frsyuki** [comment]
I think I found the cause. it's a race condition around add and finish.

**Comment by frsyuki** [comment]
#383 should fix this issue.

**Comment by sonots** [comment]
@frsyuki it seems the problem is fixed with #383. It worked fine, thanks :)
